server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -52,7 +52,9 @@ package object server {
     def apply[F[_]: Monad, T](
         authUser: Kleisli[OptionT[F, ?], Request[F], T]): AuthMiddleware[F, T] =
       service => {
-        service.compose(Kleisli((req: Request[F]) => authUser(req).map(AuthedRequest(_, req))))
+        Kleisli((r: Request[F]) => authUser(r).map(AuthedRequest(_, r)))
+          .andThen(service.mapF(o => OptionT.liftF(o.fold(Response[F](Status.NotFound))(identity))))
+          .mapF(o => OptionT.liftF(o.fold(Response[F](Status.Unauthorized))(identity)))
```

Comment:
Should these canned responses come with a default message?  I've never had a good feel for that.  It feels like yes, but then people fight about the content type, and it's a bit anglocentric.

Reply:
I'd rather they not. If people wanted this, they could use the method that deals with the explicit error in `authmiddleware`.
I'm partial to simply the 401, but I'll let others take the wheel.